controllers/quay/features.go

```diff
@@ -35,267 +33,311 @@ const (
+func (r *QuayRegistryReconciler) checkBuildManagerAvailable(
+	qctx *quaycontext.QuayRegistryContext, bundle *corev1.Secret,
+) error {
 	var config map[string]interface{}
-	if err := yaml.Unmarshal(configBundle["config.yaml"], &config); err != nil {
-		return ctx, quay, err
+	if err := yaml.Unmarshal(bundle.Data["config.yaml"], &config); err != nil {
+		return err
 	}
 
 	if buildManagerHostname, ok := config["BUILDMAN_HOSTNAME"]; ok {
-		ctx.BuildManagerHostname = buildManagerHostname.(string)
+		qctx.BuildManagerHostname = buildManagerHostname.(string)
 	}
 
-	return ctx, quay, nil
+	return nil
 }
 
 // Validates if the monitoring component can be run. We assume that we are
 // running in an Openshift environment with cluster monitoring enabled for our
 // monitoring component to work
-func (r *QuayRegistryReconciler) checkMonitoringAvailable(ctx *quaycontext.QuayRegistryContext, quay *v1.QuayRegistry, configBundle map[string][]byte) (*quaycontext.QuayRegistryContext, *v1.QuayRegistry, error) {
+func (r *QuayRegistryReconciler) checkMonitoringAvailable(
+	ctx context.Context, qctx *quaycontext.QuayRegistryContext,
+) error {
 	if len(r.WatchNamespace) > 0 {
-		msg := "monitoring is only supported in AllNamespaces mode. Disabling component monitoring"
+		msg := "Monitoring is only supported in AllNamespaces mode. Disabling."
 		r.Log.Info(msg)
-		err := fmt.Errorf(msg)
-		return ctx, quay, err
+		return fmt.Errorf(msg)
 	}
 
 	var serviceMonitors prometheusv1.ServiceMonitorList
-	if err := r.Client.List(context.Background(), &serviceMonitors); err != nil {
+	if err := r.Client.List(ctx, &serviceMonitors); err != nil {
 		r.Log.Info("Unable to find ServiceMonitor CRD. Monitoring component disabled")
-		return ctx, quay, err
+		return err
 	}
 	r.Log.Info("cluster supports `ServiceMonitor` API")
 
 	var prometheusRules prometheusv1.PrometheusRuleList
-	if err := r.Client.List(context.Background(), &prometheusRules); err != nil {
+	if err := r.Client.List(ctx, &prometheusRules); err != nil {
 		r.Log.Info("Unable to find PrometheusRule CRD. Monitoring component disabled")
-		return ctx, quay, err
+		return err
 	}
 	r.Log.Info("cluster supports `PrometheusRules` API")
 
-	namespaceKey := types.NamespacedName{Name: GrafanaDashboardConfigNamespace}
+	namespaceKey := types.NamespacedName{
+		Name: GrafanaDashboardConfigNamespace,
+	}
+
 	var grafanaDashboardNamespace corev1.Namespace
-	if err := r.Client.Get(context.Background(), namespaceKey, &grafanaDashboardNamespace); err != nil {
-		msg := fmt.Sprintf("Unable to find the Grafana config namespace %s. Monitoring component disabled", GrafanaDashboardConfigNamespace)
+	if err := r.Client.Get(ctx, namespaceKey, &grafanaDashboardNamespace); err != nil {
+		msg := fmt.Sprintf(
+			"Unable to find the Grafana config namespace %s. "+
+				"Monitoring component disabled",
+			GrafanaDashboardConfigNamespace,
+		)
 		r.Log.Info(msg)
```

Comment:
Don't we already log this as an error on the caller's side?

Reply:
Removing the log.